azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/OAuthCredentialRetriever.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core;
+
+import com.azure.identity.InteractiveBrowserCredential;
+import com.azure.identity.InteractiveBrowserCredentialBuilder;
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.tools.auth.exception.DesktopNotSupportedException;
+import com.microsoft.azure.tools.auth.exception.LoginFailureException;
+import com.microsoft.azure.tools.auth.model.AuthMethod;
+import com.microsoft.azure.tools.auth.model.AzureCredentialWrapper;
+import me.alexpanov.net.FreePortFinder;
+
+import java.awt.*;
+
+public class OAuthCredentialRetriever extends AbstractCredentialRetriever {
+    private static final String AZURE_TOOLKIT_CLIENT_ID = "777acee8-5286-4d6e-8b05-f7c851d8ed0a";
+
+    public OAuthCredentialRetriever(AzureEnvironment env) {
+        this.env = env;
+    }
+
+    public AzureCredentialWrapper retrieveInternal() throws LoginFailureException {
+        if (!isBrowserAvailable()) {
+            throw new DesktopNotSupportedException("Not able to launch a browser to log you in.");
+        }
+
+        InteractiveBrowserCredential interactiveBrowserCredential = new InteractiveBrowserCredentialBuilder()
+                .clientId(AZURE_TOOLKIT_CLIENT_ID).redirectUrl("http://localhost:" + FreePortFinder.findFreeLocalPort())
```

Comment:
In Identity SDK, user can supply their own Client ID for Interactive Browser and Device Code credential. 
In this case, i will always use Toolkit's Client id. So, user credentials will always be authorizing the toolkit public client app instead of their own.

Reply:
The  AZURE_TOOLKIT_CLIENT_ID is applied by our team, and the `auth-lib` in this PR is only used by the tools owned by our team.